File: Exemplos/App_IntentDigitalHUB_XamarinAndroid/Elgin_IntentDigitalHUB_XamarinAndroid/IntentServices/IntentDigitalHubModule.cs
namespace Xamarin_Android_Intent_Digital_Hub.IntentServices
{
    public class IntentDigitalHubModule
    {
        private IntentDigitalHubModule(string value) { Value = value; }

        public string Value { get; private set; }

        public static IntentDigitalHubModule BRIDGE = new IntentDigitalHubModule("com.elgin.e1.digitalhub.BRIDGE");
        public static IntentDigitalHubModule SAT = new IntentDigitalHubModule("com.elgin.e1.digitalhub.SAT");
        public static IntentDigitalHubModule TERMICA = new IntentDigitalHubModule("com.elgin.e1.digitalhub.TERMICA");
    }
}

File: Exemplos M10/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/IntentDigitalHubService/intent_digital_hub_module.dart
// ignore: constant_identifier_names
enum IntentDigitalHubModule { BALANCA, BRIDGE, TERMICA, SAT }

extension DigitalHubExtension on IntentDigitalHubModule {
  String get intentPath {
    switch (this) {
      case IntentDigitalHubModule.BALANCA:
        return 'com.elgin.e1.digitalhub.BALANCA';
      case IntentDigitalHubModule.BRIDGE:
        return 'com.elgin.e1.digitalhub.BRIDGE';
      case IntentDigitalHubModule.TERMICA:
        return 'com.elgin.e1.digitalhub.TERMICA';
      case IntentDigitalHubModule.SAT:
        return 'com.elgin.e1.digitalhub.SAT';
    }
  }
}
